src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -116,12 +111,20 @@ public FileResource(FileService fileService, ResourceLoaderService resourceLoade
      * @param keepFileName specifies if original file name should be kept
      * @return The path of the file
      * @throws URISyntaxException if response path can't be converted into URI
+     * @deprecated Implement your own usage of {@link FileService#handleSaveFile(MultipartFile, boolean, boolean)} with a mixed multipart request instead. An example for this is
+     * * {@link AttachmentUnitResource#updateAttachmentUnit(Long, Long, AttachmentUnit, Attachment, MultipartFile, boolean, String)}
      */
+    @Deprecated
     @PostMapping("fileUpload")
     @PreAuthorize("hasRole('TA')")
     public ResponseEntity<String> saveFile(@RequestParam(value = "file") MultipartFile file, @RequestParam(defaultValue = "false") boolean keepFileName) throws URISyntaxException {
         log.debug("REST request to upload file : {}", file.getOriginalFilename());
-        return handleSaveFile(file, keepFileName, false);
+        String responsePath = fileService.handleSaveFile(file, keepFileName, false);
```

Comment:
💡Idea: Maybe refactor implementation into own method as same as saveMarkdownFile?

Reply:
Not sure what you mean here. There is a specific endpoint for the markdown upload.

Reply:
![image](https://user-images.githubusercontent.com/29718932/187030291-b0303a6d-b8c3-4227-9b25-7b6fe50b5010.png)
![image](https://user-images.githubusercontent.com/29718932/187030292-2a87eac1-e8aa-4f8b-ba26-4c3878185079.png)
These implementations are nearly identical. I mean you could if you want extract the common part into a private helper method to reduce code duplication :)

Reply:
As discussed, I don't think it's worth doing for those few loc. The whole saving part is already extracted.